[CRD] invalid property name

#### Setup Details
CIS Version : 2.10.0
Build: f5networks/k8s-bigip-ctlr:latest
BIGIP Version: Big IP 15.1.0.5
AS3 Version: 3.22.1
Agent Mode: AS3
Orchestration: K8S
Orchestration Version: 1.21.5
Pool Mode: ClusterIP (calico bgp)
Additional Setup details: Calico

#### Description  

Unable to apply TransportServer CR with valid syntax

#### Steps To Reproduce
1) Deploy CRDs: https://raw.githubusercontent.com/F5Networks/k8s-bigip-ctlr/master/docs/config_examples/customResourceDefinitions/customresourcedefinitions.yml
2) Deploy 

```
apiVersion: "cis.f5.com/v1"
kind: TransportServer
metadata:
   name: transport-server-53-tcp
   labels:
     f5cr: "true"
spec:
  virtualServerAddress: "2001:0db8:85a3:0000:0000:8a2e:0370:7334"
  virtualServerPort: 53
  virtualServerName: dns-tcp
  mode: performance
  type: tcp
  snat: auto
  pool:
    service: dnsdist_testdist
    servicePort: 53
    monitor:
      type: tcp
      interval: 10
      timeout: 31
  serviceAddress:
  - icmpEcho: enable
    arpEnabled: true
    trafficGroup: "/Common/traffic-group-1"
```

3) Check logs for error

#### Expected Result 

Able to deploy ts CR

#### Actual Result  

error
```
2022/09/22 03:56:29 [DEBUG] [AS3] PostManager Accepted the configuration
2022/09/22 03:56:29 [DEBUG] [AS3] posting request to https://172.2.2.2/mgmt/shared/appsvcs/declare/k8s
2022/09/22 03:56:29 [ERROR] [AS3] Raw response from Big-IP: map[code:422 declarationFullId: errors:[/k8s/Shared: propertyName "crd_service_address_2001:0db8:85a3:0000:0000:8a2e:0370:7334" should match pattern "^[A-Za-z]([0-9A-Za-z_.-]{0,188}[0-9A-Za-z_.])?$"] message:declaration is invalid] {"$schema":"https://raw.githubusercontent.com/F5Networks/f5-appsvcs-extension/master/schema/3.26.0/as3-schema-3.26.0-5.json","class":"AS3","declaration":{"class":"ADC","controls":{"class":"Controls","userAgent":"CIS/v2.10.0 K8S/v1.21.5"},"id":"urn:uuid:85626792-9ee7-46bb-8fc8-4ba708cfdc1d","k8s":{"Shared":{"class":"Application","crd_service_address_2001:0db8:85a3:0000:0000:8a2e:0370:7334":{"class":"Service_Address","virtualAddress":"2001:0db8:85a3:0000:0000:8a2e:0370:7334","arpEnabled":true,"icmpEcho":"enable","trafficGroup":"/Common/traffic-group-1","spanningEnabled":false},"pl-istio-es":{"class":"Pool","monitors":[{"use":"/k8s/Shared/pl-istio-es-monitor"}]},"pl-istio-es-monitor":{"class":"Monitor","interval":10,"monitorType":"tcp","targetAddress":"","timeout":31,"adaptive":false,"receive":"","send":""},"template":"shared","vs_istio_es_8080":{"layer4":"tcp","class":"Service_L4","virtualAddresses":[{"use":"crd_service_address_2001:0db8:85a3:0000:0000:8a2e:0370:7334"}],"virtualPort":8080,"snat":"auto","pool":"pl-istio-es","profileL4":"basic"}},"class":"Tenant","defaultRouteDomain":0},"label":"CIS Declaration","remark":"Auto-generated by CIS","schemaVersion":"3.26.0"}}
2022/09/22 03:56:29 [ERROR] [AS3] Big-IP Responded with code: 422
```



#### Observations (if any)
It seems AS3 doesn't like using the below name for the service address name:

"use":"crd_service_address_2001:0db8:85a3:0000:0000:8a2e:0370:7334"

--- a/docs/RELEASE-NOTES.rst
+++ b/docs/RELEASE-NOTES.rst
@@ -9,13 +9,15 @@ Added Functionality
 
 Bug Fixes
 ````````````
+* Fix to monitor NGINX+ service changes
+* Fix issue with inconsistent pool names for VS
+* :issues:`2596` Fix invalid property name with serviceAddress
 * :issues:`2570` Fix for TLSProfile doesn't get updated when K8s secret changes
 * :issues:`2394` Fix to set ingress https monitor send string
 * :issues:`2549` Fix trafficGroup regex
-* : Fix issue with inconsistent pool names for VS
 * :issues:`2492` Fix for adding unique pool members in nodePort mode
 * :issues:`847`  Fix issue with adding both VIPs to GTM pool members for IngressLink
-* :issues: Fix to monitor NGINX+ service changes
+
 
 
 2.10.0

--- a/pkg/controller/backend.go
+++ b/pkg/controller/backend.go
@@ -1087,7 +1087,7 @@ func createServiceAddressDecl(cfg *ResourceConfig, virtualAddress string, shared
 		serviceAddress.SpanningEnabled = sa.SpanningEnabled
 		serviceAddress.TrafficGroup = sa.TrafficGroup
 		serviceAddress.VirtualAddress = virtualAddress
-		name = "crd_service_address_" + strings.Replace(virtualAddress, ".", "_", -1)
+		name = "crd_service_address_" + AS3NameFormatter(virtualAddress)
 		sharedApp[name] = serviceAddress
 	}
 	return name

